prompting/apparmor/ioctl.go

```diff
@@ -0,0 +1,83 @@
+package apparmor
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+var doSyscall = func(trap, a1, a2, a3 uintptr) (r1, r2 uintptr, err unix.Errno) {
+	return unix.Syscall(trap, a1, a2, a3)
+}
+
+type hexBuf []byte
+
+func (hb hexBuf) String() string {
+	var buf bytes.Buffer
+	for i, b := range hb {
+		fmt.Fprintf(&buf, "%#02x", b)
+		if i != len(hb)-1 {
+			fmt.Fprintf(&buf, ", ")
+		}
+		if i > 0 && (i+1)%16 == 0 {
+			fmt.Fprintf(&buf, "\n")
+		}
+	}
+	return buf.String()
+}
+
+var dumpIoctl bool = osutil.GetenvBool("SNAPD_DEBUG_DUMP_IOCTL")
+
+// NotifyIoctl performs a ioctl(2) on the apparmor .notify file.
```

Comment:
this should say a bit more about the relation of msg and the Msg* types in the packages, should even msg get a dedicated type?

Reply:
Good idea, thanks! I've added a dedicated `IoctlRequestBuffer` type, and moved the original `RequestBuffer()` function from `message.go` to `NewIoctlRequestBuffer()` in `ioctl.go`. The buffer parameter to `NotifyIoctl()` is now a `IoctlRequestBuffer`. It also modifies the size of the buffer which was passed into the syscall after checking that the return value was sane.